frontend/src/scenes/insights/insightLogic.tsx

```diff
@@ -337,7 +337,11 @@ export const insightLogic: LogicWrapper<insightLogicType> = kea<insightLogicType
                 )
             },
         ],
-        showPersonsModal: [() => [(s) => s.query], (query) => !query || !query.hidePersonsModal],
+        showPersonsModal: [
+            (s) => [s.query, s.insightProps],
+            (query: Record<string, any>, insightProps: InsightLogicProps): boolean =>
+                !(query?.hidePersonsModal || insightProps?.query?.hidePersonsModal),
+        ],
```

Comment:
I'll change my review. I'm happy to merge this *if* you can show me this is not breaking existing cases. See, previously we were returning true if `query` was false-ish, but that's not true anymore after your change.

I think this is what this should look like
```typescript
const theQuery = query || insightProps?.query
return !query || !query.hidePersonsModal
```

Reply:
I also like that. With real data, it will be the same as the query, which was always undefined. I can't confirm without any data, but I did log and add a breakpoint for `query != null`, and it was never reached.